### PR TITLE
Use v1 of GitHub action `pypa/gh-action-pypi-publish`

### PR DESCRIPTION
**Pull request by michael-k**
The `master` branch version has been sunset, see https://github.com/pypa/gh-action-pypi-publish#-master-branch-sunset-

At the moment there's no diff between `master` and `release/v1`, see https://github.com/pypa/gh-action-pypi-publish/compare/master...release/v1

##### Contributor checklist

- [x] Provided the tests for the changes.
- [x] Assure PR title is short, clear, and good to be included in the user-oriented changelog

##### Maintainer checklist

- [ ] Assure one of these labels is present: `backwards incompatible`, `feature`, `enhancement`, `deprecation`, `bug`, `dependency`, `docs` or `skip-changelog` as they determine changelog listing.
- [ ] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

**Review by atugushev — APPROVED** [review]
Thanks! 👍

**Comment by webknjaz** [comment]
> At the moment there's no diff between `master` and `release/v1`, see [pypa/gh-action-pypi-publish@master...release/v1](https://github.com/pypa/gh-action-pypi-publish/compare/master...release/v1)

That is not entirely correct :)
You're linking a "three-dot diff" performed in the wrong direction.
https://github.com/pypa/gh-action-pypi-publish/compare/master..release/v1 and https://github.com/pypa/gh-action-pypi-publish/compare/release/v1...master will correctly show the changes — two commits to be precise.
Yes, they are non-functional and only relate to showing the deprecation message but still, they are changes.

**Comment by michael-k** [comment]
> You're linking a "three-dot diff" performed in the wrong direction.

Let me rephrase: At the moment there are no new commits in `release/v1` compared to `master`.

But you're right, I should have checked the other way as well. :+1:
